Lib/test/test_concurrent_futures/test_wait.py

```diff
@@ -54,11 +58,24 @@ def test_first_completed_some_already_completed(self):
                 finished)
         self.assertEqual(set([future1]), pending)
 
-    @support.requires_resource('walltime')
+        event.set()
+        future1.result()  # wait for job to finish
+
     def test_first_exception(self):
```

Comment:
Me trying to understand it loudly:

- Schedule a normal future (fut1)
- Schedule a future (fut2) that waits until `ev1.set()` is called. When `ev1.set()` is called, raise.
- Schedule a future (fut3) that waits until `ev2.set()` is called.
- Launch a thread just to first make fut1 complete, then fut2 by setting `ev1`.
- Because fut2 is now completed, it will then raise.
- This makes `futures.wait()` exit.
- So what remains is `fut3`. We just call `ev2.set()` to make it complete and then we clean it up.

Reply:
Yes, that's right.